StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/SpacingRules/SA1023CSharp9UnitTests.cs

```diff
@@ -3,9 +3,166 @@
     {
+        [Fact]
+        public async Task TestFunctionPointerParameterValidSpacingAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerParameterInvalidPrecedingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int {|#0:*|}> FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotPreceded).WithLocation(0);
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerParameterInvalidTrailingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int{|#0:*|} > FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotFollowed).WithLocation(0);
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerTypeValidSpacingAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerTypeInvalidPrecedingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate {|#0:*|}<int*> FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotPreceded).WithLocation(0);
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerTypeInvalidTrailingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate{|#0:*|} <int*> FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotFollowed).WithLocation(0);
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerTypeWithCallingConventionValidSpacingAsync()
```

Comment:
💡 Groups of tests for variations of the same scenario can be written as a single test using the following form:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/7ae14438aac88d8c9ba1591cb21b20be3483f318/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp7/SpacingRules/SA1008CSharp7UnitTests.cs#L28-L42

This helps the reader see that all the cases are covered.

The example was implemented before the new `WithLocation` syntax was available, but you can combine both here by using `WithLocation(0)` for the first span, `WithLocation(1)` for the second, and so on.